blocks/loops.ts

```diff
@@ -325,8 +325,12 @@ export const loopTypes: Set<string> = new Set([
   'controls_whileUntil',
 ]);
 
-/** Type of a block that has CONTROL_FLOW_IN_LOOP_CHECK_MIXIN */
-type ControlFlowInLoopBlock = Block & ControlFlowInLoopMixin;
+/**
+ * Type of a block that has CONTROL_FLOW_IN_LOOP_CHECK_MIXIN
+ *
+ * @internal
```

Comment:
As discussed:

My thinking is that it is useful to make these types available so the generator functions can use them for casts (to access the extra properties), but I didn't want to make them public given that they are basically hacks I added to get the library blocks to pass type checking.

My plan remains to try to substantially improve the typing of the block definitions, probably by changing the way the mixin object are defined: Define blocks as subclasses of `Block` (or possibly in some cases subclasses of `BlockSvg`), and then using the `.prototype` object as (or to create) the mixin that is the actual block definition added to `Blockly.Blocks`.  This obviates the need to add a third mechanism for defining blocks to  `core/`, while allowing the types of things within the block definitions to be specified in a more straightforward and understandable way.

Once that's done then I think it would be reasonable to make the resulting type declarations fully public.